baselines/statavg/README.md

```diff
@@ -0,0 +1,101 @@
+
+**Datasets:** TON IoT dataset (linux memory logs). Online at https://research.unsw.edu.au/projects/toniot-datasets
+
+**Hardware Setup:**  These experiments were run on a desktop machine with 16 CPU threads. Any machine with 4 CPU cores or more would be able to run it in a reasonable amount of time.
+
+**Contributors:** TBD
+
+
+## Experimental Setup
+
+**Task:** Classification of cyberattacks.
+
+**Model:** A simple MLP with three hidden layers.
+
+**Dataset:** This baseline only includes the TON IoT dataset. By default, it will be partitioned into 5 clients following a stratified split based on the labels. The settings are as follows:
+| Dataset | #classes | #partitions | partitioning method | partition settings |
+| :------ | :---: | :---: | :---: | :---: |
+| TON IoT | 6 | 5 | stratified based on labels | 6 classes per client |
+
+**Training Hyperparameters:** The following table shows the main hyperparameters for this baseline with their default value.
+| Description | Default Value |
+| ----------- | ----- |
+| total clients | 5 |
+| clients per round | 5 |
+| number of rounds | 40 |
+| local epochs | 2 |
+| client resources | {'num_cpus': 2.0, 'num_gpus': 0.0 }|
+| data partition | stratified based on labels (6 classes per client) |
+| optimizer | Adam |
+
+
+## Environment Setup
+
+To construct the Python environment, simply run:
+
+```bash
+# Set directory to use python 3.10 (install with `pyenv install <version>` if you don't have it)
+pyenv local 3.10.13
+
+# Tell poetry to use python3.10
+poetry env use 3.10.13
+
+# Install
+poetry install
+```
+
+## Running the Experiments
+Firstly, ensure that the dataset is located at `dataset/dataset.csv`. If you want to run the experiments with your own data, place your dataset in this path.
+To run StatAvg with TON IoT baseline, ensure you have activated your Poetry environment (execute `poetry shell` from this directory), then:
+
+```bash
+python -m statavg.main # this will run using the default settings in the `conf/base.yaml`
+
+# you can override settings directly from the command line
+python -m statavg.main num_rounds=20 # will set number of rounds to 20
+```
+It is noted that an auxiliary directory is created in the runtime to persist client state, which includes the data scalers/normalizers of each client. By default, this directory is deleted upon the termination of the execution. To prevent automatic deletion (in this case, it is recommended to manually delete the directory from previous runs), execute the following:
+```bash
+# disable automated removal of the auxiliary directory of scalers
+python -m statavg.main delete_scaler_dir=False
+```
+
+## Expected Results
+
+To reproduce the results of the paper (Fig. 3., StatAvg), simply run:
+
+```bash
+python -m statavg.main   # default settings
+```
+The expected results should look similar to the following figure:
```

Comment:
Could you indicate briefly that the expected results is in the `outputs/` directory and saved into a pickle file?